File: back/central/services/integrations/messaging/whatsapp/internal/domain/entities/cod_amount_test.go
package entities

import "testing"

func TestCodAmountToCollect(t *testing.T) {
	tests := []struct {
		name                string
		codTotal            float64
		codCarrierFee       float64
		codIncludesShipping bool
		want                float64
	}{
		{"orden manual: cod_total neto, se suma la comision", 135929, 9451, false, 145380},
		{"checkout woocommerce: cod_total ya trae la comision (orden 15789)", 73367, 5365, true, 73367},
		{"sin comision conocida todavia", 73367, 0, false, 73367},
		{"sin cod_total no se inventa un monto", 0, 5365, false, 0},
	}

	for _, tt := range tests {
		t.Run(tt.name, func(t *testing.T) {
			got := CodAmountToCollect(tt.codTotal, tt.codCarrierFee, tt.codIncludesShipping)
			if got != tt.want {
				t.Errorf("CodAmountToCollect(%v, %v, %v) = %v, se esperaba %v",
					tt.codTotal, tt.codCarrierFee, tt.codIncludesShipping, got, tt.want)
			}
		})
	}
}

File: front/central/src/shared/utils/cod-amount.test.ts
import { describe, it, expect } from 'vitest';
import { codCustomerCharge, codBusinessNet } from './cod-amount';

describe('cod-amount', () => {
    it('orden del checkout de WooCommerce: el cod_total ya trae la comision (orden 15789)', () => {
        const order = { cod_total: 73367, cod_carrier_fee: 5365, cod_includes_shipping: true };
        expect(codCustomerCharge(order)).toBe(73367);
        expect(codBusinessNet(order)).toBe(68002);
    });

    it('orden manual: la comision se suma al cod_total (VIG-0069)', () => {
        const order = { cod_total: 135929, cod_carrier_fee: 9451, cod_includes_shipping: false };
        expect(codCustomerCharge(order)).toBe(145380);
        expect(codBusinessNet(order)).toBe(135929);
    });

    it('sin comision conocida todavia muestra el cod_total', () => {
        expect(codCustomerCharge({ cod_total: 73367, cod_carrier_fee: 0, cod_includes_shipping: false })).toBe(73367);
    });

    it('sin cod_total no inventa un monto', () => {
        expect(codCustomerCharge({ cod_total: 0, cod_carrier_fee: 5365 })).toBe(0);
        expect(codBusinessNet({ cod_carrier_fee: 5365 })).toBe(0);
    });
});
